helper-script

```diff
@@ -0,0 +1,270 @@
+    >&2 echo "Need the steam account name to update. (-n name)"
+    exit 1
+fi
+
+# only one game at a time
+if [ "$ats" = true ] && [ "$ets2" = true ]; then
+    >&2 echo "It's only possible to use one game at a time."
+    exit 1
+elif ! [ "$ats" = true ] && ! [ "$ets2" = true ]; then
+    >&2 echo "Need at least one game. ATS (-a) or ETS2 (-e)"
+    exit 1
+elif [ "$ats" = true ]; then
+    steamid="270880"
+    if [ -z "$prefixdir" ]; then
+        prefixdir="$XDG_DATA_HOME/truckersmp-cli/American Truck Simulator/prefix"
+    fi
+    if [ -z "$gamedir" ]; then
+        gamedir="$XDG_DATA_HOME/truckersmp-cli/American Truck Simulator/data"
+    fi
+elif [ "$ets2" = true ]; then
+    steamid="227300"
+    if [ -z "$prefixdir" ]; then
+        prefixdir="$XDG_DATA_HOME/truckersmp-cli/Euro Truck Simulator 2/prefix"
+    fi
+    if [ -z "$gamedir" ]; then
+        gamedir="$XDG_DATA_HOME/truckersmp-cli/Euro Truck Simulator 2/data"
+    fi
+fi
+
+echo "Using $steamid"
+echo "GAMEDIR: $gamedir"
+echo "prefixdir: $prefixdir"
+echo "protondir: $protondir"
+
+# check $protondir
+if [ "$startgame" = true ] &&  ! [ -f "$protondir/proton" ]; then
+    >&2 echo "Proton not found."
+    >&2 echo    "Make sure to start at least one game with proton manually or" \
+                "set the correct path with -p"
+    exit 1
+fi
+
+# check for game if no update
+if  ! [ "$update" = true ] \
+    && ! [ -f "$gamedir/bin/win_x64/eurotrucks2.exe" ] \
+    && ! [ -f "$gamedir/bin/win_x64/amtrucks.exe" ]; then
+    >&2 echo "No update wanted and game not found."
+    exit 1
+fi
+}
+
+# download ATS/ETS2
+if [ "$update" = true ]; then
+    # make sure steam is closed before updating
+    # it's possible to update with the steam client open but the client looses
+    # all connectivity and asks for password and steam guard code after restart
+    while pidof steam; do
+	    pkill steam
+	    sleep 5
+    done
+
+    # get steamcmd if not already available
+    if ! [ -f "$moddir/steamcmd/steamcmd.sh" ]; then
+        mkdir -p "$moddir/steamcmd"
+        wget -qO- $steamcmdurl | tar xvz -C "$moddir/steamcmd"
+    fi
+
+    if ! [ -d "$gamedir" ]; then
+        mkdir -p "$gamedir"
+    fi
+
+    "$moddir/steamcmd/steamcmd.sh" \
+        +@sSteamCmdForcePlatformType windows \
+        +login "$accountname" \
+        +force_install_dir "$gamedir" \
+        +app_update "$steamid" validate \
+        +quit
+
+    # update the script itself
+    if git -C "$scriptdir" rev-parse --is-inside-work-tree > /dev/null 2>&1; then
+        git -C "$scriptdir" pull origin master
+    fi
+fi
+
+# always update mod
+"$scriptdir/truckersmp-cli" -u -d "$moddir"
+
+# start with Proton
+if [ "$startgame" = true ]; then
+    # make sure steam is started
+    if ! pidof steam; then
+        nohup steam > /dev/null 2>&1 &
+
+        # maybe wait a bit for steam startup
+        # steam is heavy
+        echo "Please wait a bit for steam to startup…"
+        sleep 60
```

Comment:
That's... extreme. Isn't there something we can wait for ? Maybe the spawning of a steam process ?

Reply:
I'm not sure how to detect if Steam is started completely. I came up with some window title search for "Steam" with `xdotool`. But this will get triggered already when the little connecting to account window is showing up. But it somehow worked in a little test.
We can't just look for the process because that's available right after we're starting it.

Reply:
I got something better. When the user gets logged in an updated timestamp in `$XDG_DATA_HOME/Steam/config/loginusers.vdf` gets written to the file. We can wait for that file change with `inotifywait` which is included in `inotify-tools`.